langfuse/_task_manager/media_manager.py

```diff
@@ -0,0 +1,245 @@
+
+from langfuse.api import GetMediaUploadUrlRequest, PatchMediaBody
+from langfuse.api.core import ApiError
+from langfuse.api.client import FernLangfuse
+from langfuse.media import LangfuseMedia
+from langfuse.utils import _get_timestamp
+
+from .media_upload_queue import MediaUploadQueue, UploadMediaJob
+
+
+T = TypeVar("T")
+P = ParamSpec("P")
+
+
+class MediaManager:
+    _log = logging.getLogger(__name__)
+
+    def __init__(
+        self,
+        *,
+        api_client: FernLangfuse,
+        media_upload_queue: MediaUploadQueue,
+        max_retries: Optional[int] = 3,
+    ):
+        self._api_client = api_client
+        self._queue = media_upload_queue
+        self._max_retries = max_retries
+
+    def process_next_media_upload(self):
+        try:
+            upload_job = self._queue.get(block=True, timeout=1)
+            self._log.debug(f"Processing upload for {upload_job['media_id']}")
+            self._process_upload_media_job(data=upload_job)
+
+            self._queue.task_done()
+        except Empty:
+            self._log.debug("Media upload queue is empty")
+            pass
+        except Exception as e:
+            self._log.error(f"Error uploading media: {e}")
+            self._queue.task_done()
+
+    def process_media_in_event(self, event: dict):
+        try:
+            if "body" not in event:
+                return
+
+            body = event["body"]
+            trace_id = body.get("traceId", None) or (
+                body.get("id", None)
+                if "type" in event and "trace" in event["type"]
+                else None
+            )
+
+            if trace_id is None:
+                raise ValueError("trace_id is required for media upload")
+
+            observation_id = (
+                body.get("id", None)
+                if "type" in event
+                and ("generation" in event["type"] or "span" in event["type"])
+                else None
+            )
+
+            multimodal_fields = ["input", "output", "metadata"]
+
+            for field in multimodal_fields:
+                if field in body:
+                    processed_data = self._find_and_process_media(
+                        data=body[field],
+                        trace_id=trace_id,
+                        observation_id=observation_id,
+                        field=field,
+                    )
+
+                    body[field] = processed_data
+
+        except Exception as e:
+            self._log.error(f"Error processing multimodal event: {e}")
+
+    def _find_and_process_media(
+        self,
+        *,
+        data: Any,
+        trace_id: str,
+        observation_id: Optional[str],
+        field: str,
+    ):
+        seen = set()
+        max_levels = 10
+
+        def _process_data_recursively(data: Any, level: int):
+            if id(data) in seen or level > max_levels:
+                return data
+
+            seen.add(id(data))
+
+            if isinstance(data, LangfuseMedia):
```

Comment:
In the openai integration, we specifically add langfusemedia for audio. This is handled here, correct?